src/sync.rs

```diff
@@ -224,19 +212,10 @@ where
 
         loop {
             match self.queue.recv() {
-                Some(SyncContextProtocol::Stop) => {
-                    self.state = ActorState::Stopping;
-                    if A::stopping(&mut act, self) != Running::Stop {
-                        warn!("stopping method is not supported for sync actors");
-                    }
-                    self.state = ActorState::Stopped;
-                    A::stopped(&mut act, self);
-                    return;
-                }
-                Some(SyncContextProtocol::Envelope(mut env)) => {
+                Ok(mut env) => {
                     env.handle(&mut act, self);
                 }
-                None => {
+                Err(_) => {
```

Comment:
Just to be sure, there is no way to detect that error is due to disconnect?

Reply:
I looked myself into it and it seems not, but since the only error case here is likely due to dropping all Senders I don't think it is likely for other error to happen